services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -235,36 +215,45 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
 
   }
 
-  override def getFile(project: Project, path: Path, version: Option[PipelineVersion])(
-    implicit ec: ExecutionContext
-  ): AsyncResult[ProjectFile] = {
-    val filePath: String = URLEncoderUtils.encode(path.toString)
-    val fileVersion: String = version match {
-      case Some(version) => version.name
-      case None          => getDefaultProjectVersion().toString
+  private def getUpdatedProjectVersion(
+    project: Project,
+    optionUserVersion: Option[PipelineVersion]
+  ): AsyncResult[PipelineVersion] =
+    getLastProjectVersion(project).map(projectVersion => getNewProjectVersion(projectVersion, optionUserVersion))
+
+  private def getLastProjectVersion(project: Project): Future[Option[PipelineVersion]] =
```

Comment:
Can't grasp, how project.version differs from the returned version. How they relate? Can `project` contain outdated version? If they are not related, maybe something with the naming can be done.

Reply:
I'm curious about it as well. `project.version` should have the latest version, so this method should not exist anymore.\
But changing behavior of `ProjectVersioning` is out of scope of current PR and I tried to change as little as possible.